Source/WebCore/svg/SVGAnimationElement.cpp

```diff
@@ -403,10 +404,12 @@ unsigned SVGAnimationElement::calculateKeyTimesIndex(float percent) const
     const auto& keyTimes = this->keyTimes();
     unsigned index;
     unsigned keyTimesCount = keyTimes.size();
-    // Compare index + 1 to keyTimesCount because the last keyTimes entry is
-    // required to be 1, and percent can never exceed 1; i.e., the second last
-    // keyTimes entry defines the beginning of the final interval
-    for (index = 1; index + 1 < keyTimesCount; ++index) {
+    // For linear and spline animations, the last value must be '1'. In those
+    // cases we don't need to consider the last value, since |percent| is never
+    // greater than one.
+    if (keyTimesCount && calcMode() != CalcMode::Discrete)
+        keyTimesCount--;
```

Comment:
We prefer `--keyTimesCount`

Reply:
Will do and then push it after last check working by api-mac. :-)

Reply:
Done and all tests pass. :-)